operations/def.go

```diff
@@ -0,0 +1,115 @@
+//  Copyright 2021, athena-crdt authors.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package operations
+
+import (
+	"sort"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/athena-crdt/athena-core/operations/defs"
+	"github.com/athena-crdt/athena-core/operations/lamport"
+)
+
+type (
+	// Mutation is the type of operation
+	Mutation uint8
+
+	// Operation - individual operation entity
+	Operation struct {
+		Id       *lamport.Clock
```

Comment:
why are we using pointers here?

Reply:
To prevent it from unnecessary copying and to have the atomicity while checking and updating the counter value.

Reply:
at operation level the clock is fixed as is used as an id so no updates. 
Suggestion:
don't use pointers here

rest looks good

Reply:
It hardly makes any difference. If you insist - value then.